pkg/storage/bloom/v1/bloom_tester_test.go

```diff
@@ -0,0 +1,155 @@
+			query:       `{app="fake"} |= "nope"`,
+			bloom:       fakeBloom{"foo", "bar"},
+			expectMatch: false,
+		},
+		{
+			name:        "two filters",
+			query:       `{app="fake"} |= "foo" |= "bar"`,
+			bloom:       fakeBloom{"foo", "bar"},
+			expectMatch: true,
+		},
+		{
+			name:        "two filters no match",
+			query:       `{app="fake"} |= "foo" |= "nope"`,
+			bloom:       fakeBloom{"foo", "bar"},
+			expectMatch: false,
+		},
+		{
+			name:        "notEq match",
+			query:       `{app="fake"} != "nope"`,
+			bloom:       fakeBloom{"foo", "bar"},
+			expectMatch: true,
+		},
+		{
+			name:        "notEq no match",
+			query:       `{app="fake"} != "foo"`,
+			bloom:       fakeBloom{"foo", "bar"},
+			expectMatch: false,
+		},
+		{
+			name:        "or filter both match",
+			query:       `{app="fake"} |= "foo" or "bar"`,
+			bloom:       fakeBloom{"foo", "bar"},
+			expectMatch: true,
+		},
+		{
+			name:        "or filter one right match",
+			query:       `{app="fake"} |= "nope" or "foo"`,
+			bloom:       fakeBloom{"foo", "bar"},
+			expectMatch: true,
+		},
+		{
+			name:        "or filter one left match",
+			query:       `{app="fake"} |= "foo" or "nope"`,
+			bloom:       fakeBloom{"foo", "bar"},
+			expectMatch: true,
+		},
+		{
+			name:        "or filter no match",
+			query:       `{app="fake"} |= "no" or "nope"`,
+			bloom:       fakeBloom{"foo", "bar"},
+			expectMatch: false,
+		},
+		{
+			name:        "Not or filter match",
+			query:       `{app="fake"} != "nope" or "no"`,
+			bloom:       fakeBloom{"foo", "bar"},
+			expectMatch: true,
+		},
+		{
+			name:        "Not or filter right no match",
+			query:       `{app="fake"} != "nope" or "bar"`,
+			bloom:       fakeBloom{"foo", "bar"},
+			expectMatch: false,
+		},
+		{
+			name:        "Not or filter left no match",
+			query:       `{app="fake"} != "foo" or "nope"`,
+			bloom:       fakeBloom{"foo", "bar"},
+			expectMatch: false,
+		},
+		{
+			name:        "Not or filter no match",
+			query:       `{app="fake"} != "foo" or "bar"`,
+			bloom:       fakeBloom{"foo", "bar"},
+			expectMatch: false,
+		},
+		{
+			name:        "complex filter match",
+			query:       `{app="fake"} |= "foo" |= "bar" or "baz" |= "fuzz" or "not" != "nope" != "no" or "none"`,
+			bloom:       fakeBloom{"foo", "bar", "baz", "fuzz"},
+			expectMatch: true,
+		},
+		// TODO: test regexes
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			expr, err := syntax.ParseExpr(tc.query)
+			assert.NoError(t, err)
+			filters := extractLineFilters(expr)
+
+			bloomTests := FiltersToBloomTest(fakeNgramBuilder{}, filters...)
+
+			assert.Equal(t, tc.expectMatch, bloomTests.Test(tc.bloom))
+		})
+	}
+}
+
+func extractLineFilters(expr syntax.Expr) (lineFilters []syntax.LineFilterExpr) {
+	visitor := &syntax.DepthFirstTraversal{
+		VisitLineFilterFn: func(v syntax.RootVisitor, e *syntax.LineFilterExpr) {
+			lineFilters = append(lineFilters, *e)
+		},
+	}
+	expr.Accept(visitor)
+	return lineFilters
+}
```

Comment:
I think you can use the function from the `syntax` package, right?

Reply:
Indeed, leftover from before rebasing